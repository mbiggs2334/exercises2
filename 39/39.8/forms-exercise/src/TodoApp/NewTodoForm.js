import React, { useState } from 'react';

const NewTodoForm = ({removeTodo, addTodo}) => {
    const [input, setInput] = useState('');

    const handleChange = (e) => {
        const {value} = e.target;
        setInput(value)
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        addTodo(input)
        setInput('');
    }

    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="todo">Todo: </label>
            <input
                type="text"
                id="todo"
                required
                placeholder="Enter a todo..."
                value={input}
                name="todo"
                onChange={handleChange}
            />
            <button>Add Todo</button>
        </form>
    )
};

export default NewTodoForm;