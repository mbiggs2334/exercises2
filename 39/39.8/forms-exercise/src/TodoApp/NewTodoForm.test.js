import React from "react";
import { render } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';
import NewTodoForm from "./NewTodoForm";

it('renders without crashing', () => {
    render(<NewTodoForm />);
});

it('matches the snapshot', () => {
    const { asFragment } = render(<NewTodoForm />);
    expect(asFragment()).toMatchSnapshot();
});