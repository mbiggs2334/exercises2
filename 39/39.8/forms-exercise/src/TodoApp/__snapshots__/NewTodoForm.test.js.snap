// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="todo"
    >
      Todo: 
    </label>
    <input
      id="todo"
      name="todo"
      placeholder="Enter a todo..."
      required=""
      type="text"
      value=""
    />
    <button>
      Add Todo
    </button>
  </form>
</DocumentFragment>
`;
